spring-kotlin/aws-ssh-server-manager/src/main/kotlin/dev/santos/awssshservermanager/lib/aws/config/AwsSsmConfig.kt

```diff
@@ -0,0 +1,44 @@
+package dev.santos.awssshservermanager.lib.aws.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.ssm.SsmClient
+import java.net.URI
+
+@Configuration
+class AwsSsmConfig {
+    @Value("\${aws.ssm.region}")
+    private val regionStr: String? = null
+
+    @Value("\${aws.ssm.endpointUri}")
+    private val endpointUriStr: String? = null
+
+    @Value("\${aws.ssm.parameterPrefix}")
+    val parameterPrefix: String? = null
+
+    @Bean
+    fun parameterPrefix(): String {
+        return when (parameterPrefix) {
+            null -> ""
+            else -> parameterPrefix!!
+        }
+    }
+
+    @Bean
+    fun ssmClient(): SsmClient {
+        val region = Region.of(regionStr)
+        return when (endpointUriStr) {
+            null -> SsmClient
+                        .builder()
+                        .region(region)
+                        .build()
+            else -> SsmClient
+                        .builder()
+                        .region(region)
+                        .endpointOverride(URI(endpointUriStr))
+                        .build()
```

Comment:
How do you test these chained things? Seems to be a java pattern, but I don't understand the advantage over the patterns used in typescript and ruby:
```typescript
const client = new SsmClient({ region: region, endpointOverride: endpointUriStr });
```

Reply:
It looks like it's forcing to do some code duplication as well

Reply:
I can make it DRYer for sure, but there is no way to avoid the code repetition

That is the kind of thing I just test against an environment or docker-compose generated environment

No need to do that at the unit level, though it would be possible, but provides little confidence

Reply:
I'm going to delete the code, but it would look something like this:
```
  fun ssmClient(): SsmClient {
    val builder = SsmClient.builder()

    if (!endpointUri.isNullOrBlank()) {
      builder.endpointOverride(URI(endpointUri))
    }

    return builder.region(Region.of(region))
      .build()
  }
```

Reply:
Also, this is called builder pattern. Quite common in the JVM/typed languages world
That dictionary of params (kwargs in Python) has not been a common practice
Though it can be achieved with a `data class` as a parameter. But data classes are recent enough and the builder pattern is widely spread atm